betty/extension/webpack/__init__.py

```diff
@@ -0,0 +1,171 @@
+"""
+Integrate Betty with `Webpack <https://webpack.js.org/>`_.
+
+This module is internal.
+"""
+
+from __future__ import annotations
+
+from asyncio import to_thread
+from collections.abc import Callable, Sequence
+from pathlib import Path
+from shutil import copytree
+from typing import Any
+
+from aiofiles.tempfile import TemporaryDirectory
+
+from betty._npm import NpmRequirement
+from betty.app.extension import Extension
+from betty.extension.webpack.build import Builder, NpmUnavailable
+from betty.extension.webpack.jinja2.filter import FILTERS
+from betty.fs import ROOT_DIRECTORY_PATH
+from betty.generate import Generator, GenerationContext
+from betty.html import CssProvider
+from betty.jinja2 import Jinja2Provider
+from betty.job import Context
+from betty.locale import Str
+from betty.requirement import Requirement, AllRequirements, AnyRequirement
+
+PREBUILT_ASSETS_DIRECTORY_PATH = ROOT_DIRECTORY_PATH / "prebuild" / "webpack"
```

Comment:
We MUST use build IDs within this directory. Otherwise we can never ensure that when looking for prebuilt assets, they are the ones we are looking for.